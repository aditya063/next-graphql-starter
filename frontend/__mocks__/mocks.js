export const user = {
  id: '1',
  email: 'a@a.com'
};

export const idea = {
  id: '1',
  content: 'mock content'
};

export const newIdea = {
  id: '1',
  content: 'new mock content'
};

export const users = [user];
export const ideas = [user];
