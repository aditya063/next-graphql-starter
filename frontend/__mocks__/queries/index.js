export {
  MOCK_IS_LOGGED_IN,
  MOCK_CURRENT_USER,
  MOCK_ERROR_CURRENT_USER,
  MOCK_SIGN_UP,
  MOCK_SIGN_IN,
  MOCK_SIGN_OUT,
  MOCK_REQUEST_RESET,
  MOCK_RESET_PASSWORD
} from './user';

export {
  MOCK_CREATE_IDEA,
  MOCK_UPDATE_IDEA,
  MOCK_DELETE_IDEA
} from './idea';
