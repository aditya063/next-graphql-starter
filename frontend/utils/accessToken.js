import fetch from 'isomorphic-unfetch';
// import cookie from 'cookie';
// import jwt from 'jsonwebtoken';

let accessToken = '';

export const setAccessToken = token => (accessToken = token);

export const getAccessToken = () => accessToken;

export const clearAccessToken = () => (accessToken = '');

export const fetchAccessToken = async refreshToken => {
  const url =
    process.env.NODE_ENV === 'development'
      ? process.env.DEV_REFRESH_URL
      : process.env.PROD_REFRESH_URL;

  const options = {
    method: 'GET',
    credentials: 'include',
    headers: { cookie: `rt=${refreshToken}` }
  };

  try {
    const response = await fetch(url, options);

    const data = await response.json();

    return data.accessToken;
  } catch (error) {
    if (process.env.NODE_ENV === 'development')
      console.error('fetchAccessToken error: ', error);
  }
};

// export const verifyAccessToken = accessToken => {
//   try {
//     jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET);
//   } catch (error) {
//     if (process.env.NODE_ENV === 'development')
//       console.error('fetchAccessToken error: ', error);
//   }
// };

// const checkAccessToken = async cookies => {
//   const accessToken = getAccessToken();

//   if (!accessToken) return true;

//   const parsedCookies = cookie.parse(cookies);

//   if (!parsedCookies.rt) return '';

//   // const response = await fetchAccessToken(parsedCookies.rt);

//   // const data = await response.json();

//   // setAccessToken(data.accessToken);
//   // setAccessTokenExpiry(data.exp);

//   // return data.accessToken;
// };
