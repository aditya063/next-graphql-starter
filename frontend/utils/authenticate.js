import jwt from 'jsonwebtoken';

const authenticate = cookie => {
  const secret = process.env.JWT_SECRET;
  const token = cookie.slice(6);

  try {
    jwt.verify(token, secret);

    return true;
  } catch {
    return false;
  }
};

const setCache = (cache, isAuthenticated) => {
  cache.writeData({
    data: {
      isLoggedIn: isAuthenticated
    }
  });
};

export default (cache, cookie) => {
  const isAuthenticated = cookie ? authenticate(cookie) : false;

  setCache(cache, isAuthenticated);
};
