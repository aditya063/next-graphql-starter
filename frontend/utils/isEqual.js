// compare props

export default (prevProps, nextProps) => {
  const obj1 = prevProps;
  console.log('TCL: isEqual -> obj1', obj1);
  const obj2 = nextProps;
  console.log('TCL: isEqual -> obj2', obj2);

  const diff = Object.keys(obj1).reduce((result, key) => {
    if (!Object.prototype.hasOwnProperty.call(obj2, key)) {
      result.push(key);
    } else if (_.isEqual(obj1[key], obj2[key])) {
      const resultKeyIndex = result.indexOf(key);

      result.splice(resultKeyIndex, 1);
    }

    return result;
  }, Object.keys(obj2));

  console.log('TCL: areEqual -> diff', diff);

  const areEqual = !diff;

  console.log('TCL: areEqual -> areEqual', areEqual);

  return areEqual;
};
