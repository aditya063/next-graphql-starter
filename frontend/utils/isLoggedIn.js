/* eslint-disable no-console */
import * as query from '../components/wrappers/WithUser/WithUser.query';

export const isLoggedIn = async client => {
  try {
    const res = await client.query({
      query: query.ME_QUERY
    });

    return res.data.me;
  } catch (err) {
    return null;
  }
};
