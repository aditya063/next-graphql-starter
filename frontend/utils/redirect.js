import Router from 'next/router';

export const redirect = (context, target) => {
  if (context.res) {
    context.res.writeHead(303, { Location: target });
    context.res.end();
  } else {
    Router.replace(target);
  }
};
