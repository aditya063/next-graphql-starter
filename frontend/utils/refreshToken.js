import jwt from 'jsonwebtoken';
import { devConErr } from './devLog';
import { refreshTokenSecret } from '../constants';

export const verifyRefreshToken = refreshToken => {
  try {
    jwt.verify(refreshToken, refreshTokenSecret);
  } catch (error) {
    devConErr(['Refresh token verify error: ', error]);
  }
};
