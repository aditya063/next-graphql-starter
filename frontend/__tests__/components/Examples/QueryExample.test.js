import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import toJson from 'enzyme-to-json';
import QueryExample, { USERS_QUERY } from '../../../components/Examples/QueryExample';
import { load } from '../../../utils/tests';
import * as mock from '../../../__mocks__/mockData';

describe('QueryExample', () => {
  let mockProps;
  let mockQueries;
  let mockMutations;
  let wrapper;

  beforeEach(() => {
    mockProps = {};
    mockQueries = [
      {
        request: { query: USERS_QUERY },
        result: {
          data: {
            users: [mock.user]
          }
        }
      }
    ];
    mockMutations = [];
    wrapper = mount(
      <MockedProvider mocks={[...mockQueries, ...mockMutations]}>
        <QueryExample {...mockProps} />
      </MockedProvider>
    );
  });

  it('matches snapshot while loading', async () => {
    expect(wrapper.text()).toContain('Loading...');

    const queryExample = toJson(wrapper.find('QueryExample'));
    expect(queryExample).toMatchSnapshot();
  });

  it('matches snapshot when loaded', async () => {
    await load(wrapper);

    expect(wrapper.text()).not.toContain('Loading...');
    const queryExample = toJson(wrapper.find('QueryExample'));
    expect(queryExample).toMatchSnapshot();
  });

  it('renders an li for each user', async () => {
    await load(wrapper);

    console.log(wrapper.debug());
    const li = wrapper.find('ul');

    expect(li).toHaveLength(1);
    expect(li.text()).toEqual(mock.user.name);
  });
});
