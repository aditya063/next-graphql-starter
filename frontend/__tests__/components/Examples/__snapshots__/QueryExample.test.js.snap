// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryExample matches snapshot when loaded 1`] = `
<QueryExample>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "USERS_QUERY",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "users",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 65,
          "start": 0,
        },
      }
    }
  >
    <ul>
      <li
        key="1"
      >
        Larry
      </li>
    </ul>
  </Query>
</QueryExample>
`;

exports[`QueryExample matches snapshot while loading 1`] = `
<QueryExample>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "USERS_QUERY",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "users",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 65,
          "start": 0,
        },
      }
    }
  >
    <p>
      Loading...
    </p>
  </Query>
</QueryExample>
`;
